File: reactive_architecture/src/reaccao/Reaccao.java
package reaccao;

public class Reaccao implements Comportamento{
	
	Estimulo estimulo;
	Accao accao;
	
	public Reaccao(Estimulo estimulo, Accao resposta) {
		this.accao = resposta;
		this.estimulo = estimulo;
		
	}
	
	public Accao activar (Estimulo estimulo) {
		return (this.estimulo == estimulo) ? this.accao:null;
			

	}

}

File: markov_reinforcement_learning/iasa-cp3/src/lib/ecr/reaccao.py
from ecr.comportamento import Comportamento

class Reaccao(Comportamento):

	def _detectar_estimulo(percepcao):
		raise NotImplementedError

	def _gerar_resposta(estimulo):
		raise NotImplementedError

	#self para atributos e métodos da própria instância
	def activar(self, percepcao):
		estimulo = self._detectar_estimulo(percepcao)
		if(estimulo != None and estimulo != False):#pode ser estimulo de objecto ou booleano de existir ou não
			return self._gerar_resposta(estimulo)

